storage/storage_dest.go

```diff
@@ -1295,6 +1292,9 @@ func (s *storageImageDestination) CommitWithOptions(ctx context.Context, options
 		if err != nil {
 			return fmt.Errorf("reading image %q: %w", intendedID, err)
 		}
+		// FIXME: With intendedID typically based on s.manifestDigest, this will fail if we ever fetch the same layer two different ways (with two different layer IDs
+		// per trustedLayerIdentityData), e.g. if we see the layer/images across graph drivers which do/don’t support partial pulls.
+		// The only way to recover would be to remove the other image, and that might not be possible.
```

Comment:
⚠️

Reply:
FWIW this also completely rules out the capability to make partial pulls _optional_. If we want a single view of each image, we have to actually only produce a single view.